_test/test_sqw/test_equal_to_tol.m

```diff
@@ -0,0 +1,227 @@
+
+        % Replace sqw objects' npix and pix arrays
+        original_sqw.data.npix = npix;
+        original_sqw.data.pix = pix;
+        shuffled_sqw = copy(original_sqw);
+        shuffled_sqw.data.pix = shuffled_pix;
+
+        assertFalse(equal_to_tol(shuffled_sqw, original_sqw, 'reorder', false));
+    end
+
+    function test_the_same_sqw_objects_are_equal_with_paged_pix(obj)
+        sqw_copy = obj.sqw_2d_paged;
+        assertTrue(equal_to_tol(obj.sqw_2d_paged, sqw_copy));
+    end
+
+    function test_the_same_sqw_objs_eq_if_fraction_of_pix_and_pix_paged(obj)
+        sqw_copy = obj.sqw_2d_paged;
+        assertTrue(equal_to_tol(obj.sqw_2d_paged, sqw_copy, 'fraction', 0.5));
+    end
+
+    function test_paged_and_non_paged_version_of_same_sqw_file_are_equal(obj)
+        assertTrue(equal_to_tol(obj.sqw_2d_paged, obj.sqw_2d, 'fraction', 0.5));
+    end
+
+    function test_false_returned_if_NaNs_in_sqw_and_nan_equal_is_false(obj)
+        original_sqw = copy(obj.sqw_2d);
+        original_sqw.data.s(5) = nan;
+        sqw_copy = copy(original_sqw);
+
+        assertFalse(equal_to_tol(original_sqw, obj.sqw_2d, 'nan_equal', false));
+        assertFalse(equal_to_tol(sqw_copy, original_sqw, 'nan_equal', false));
+    end
+
+    function test_true_returned_if_NaNs_in_sqw_and_nan_equal_is_true(obj)
+        original_sqw = copy(obj.sqw_2d);
+        original_sqw.data.s(5) = nan;
+        sqw_copy = copy(original_sqw);
+
+        assertFalse(equal_to_tol(original_sqw, obj.sqw_2d, 'nan_equal', true));
+        assertTrue(equal_to_tol(sqw_copy, original_sqw, 'nan_equal', true));
+    end
+
+    function test_using_fraction_argument_only_tests_fraction_of_the_bins(obj)
+        original_sqw = copy(obj.sqw_2d_paged);
+        npix = [10, 5, 6, 3, 6];
+        bin_end_idxs = cumsum(npix);
+        bin_start_idxs = bin_end_idxs - npix + 1;
+
+        fraction = 0.5;
+
+        data = rand(PixelData.DEFAULT_NUM_PIX_FIELDS, 30);
+        edited_data = data;
+        bins_to_edit = round(2:(1/fraction):numel(npix));
+
+        for i = 1:numel(bins_to_edit)
+            bin_num = bins_to_edit(i);
+            edited_data(:, bin_start_idxs(bin_num):bin_end_idxs(bin_num)) = 0;
+        end
+
+        npix_in_page = 11;
+        pix = obj.get_pix_with_fake_faccess(data, npix_in_page);
+        edited_pix = obj.get_pix_with_fake_faccess(edited_data, npix_in_page);
+
+        % Replace sqw objects' npix and pix arrays
+        original_sqw.data.npix = npix;
+        original_sqw.data.pix = pix;
+        edited_sqw = copy(original_sqw);
+        edited_sqw.data.pix = edited_pix;
+
+        assertFalse(equal_to_tol(edited_sqw, original_sqw, 'fraction', 1));
+        assertTrue(equal_to_tol(edited_sqw, original_sqw, 'fraction', fraction));
+    end
+
+    function test_using_fraction_argument_is_faster_than_comparing_all_pix(obj)
+        sqw_copy = copy(obj.sqw_2d);
+
+        num_reps = 5;
+        num_iters = 1;
```

Comment:
having created the `num_iters` argument to the benchmark it is a shame to set it to one the only time it's called

Reply:
I originally added the argument so I could tweak both parameters and see what gave stable results.

The `niters` argument is really intended to be used for functions where random noise is comparable to time taken, that doesn't seem to be the case here.